### DISCUSSION TITLE
Pressing left/right with something typed moves text cursor instead of the entry selector, breaking multiple-column configs.

### DISCUSSION THREAD

**Discussion by Asmageddon** [davatorium/rofi#1705] [(Possible) Issues]
This appears to be a deliberate change, but it's extremely problematic for dmenu-style configs, and IMO not a sane default for multiple-column configs overall.

**Reproduction**:
 * Open rofi
 * Press <kbd>right</kbd> to move to the entry on the right
 * Type anything into the search field
 * Press <kbd>left</kbd>

**Expected behavior**: Moving to the entry on the left of current.
**Outcome**: The text entry cursor is moved instead.

**Comment by DaveDavenport** [comment]
how do you expect to move in your textbox then? This was added on request.
But you can rebind all the keybindings, I think you can change this.

**Reply by DaveDavenport** [reply]
see `rofi -show keys`   unbind left/right and map that to element next/prev.

**Reply by Asmageddon** [reply]
Honestly, I don't think moving in the textbox is actually useful for searching through available options. If you make a typo, it's usually obvious the moment you make it, or 1-2 characters back.

Honestly, the only usecase where moving the cursor in the textbox seems like a saner default is typing in commands for `run` modi, which feels to me is a pretty niche usecase compared to using rofi as an application launcher where you type 3-4 characters, maybe press a few arrow keys, and press enter.

Furthermore, this update breaks a lot of existing configs since rofi refuses to launch with overlapping keybindings, so if you defined any previously... well tough luck.

I believe the change to defaults should be rolled back, with this style of use documented as a copy-paste example instead.

**Reply by DaveDavenport** [reply]
1. Honestly, this behavior has been there for a very long time.  It was requested to extend the cursor moving to move the selection when hitting the ends of the entry text.   I am not going to change this behaviour now, it is generally not appreciated if I change default behaviour without discussion or good reason.
2. Honestly, see 1.
3. No, overlapping keybinding always generated errors since we moved to xkbcommon many years ago. This is not a new thing.  What has been added was more a flexible flow to the listview, you can now set a 'flow direction'.
This was also often requested feature by users who make fancy scripts around users (f.e. for adding a calendar view).  This new flow added two extra keybindings. After a discussion with users it was decided binding these new keybindings by default was worth the effort and slight inconvenient to the few users that already bound those keys. This is not something done without considering the consequences. 

During the development up to the release, no user complained about this. If you want to help make releases better, please consider testing the development version and give direct feedback.

**Reply by Asmageddon** [reply]
Then I must have somehow had an old default config on my system, because for me left/right moved the selection without affecting the text all the way since 2021.05 until today's update from 1.7.3 to 1.7.5. Which IMO was a much saner default for the standard usecase of launching programs.

**Reply by DaveDavenport** [reply]
left/right i bound to `move-char-back` and `move-char-forward`.  This has been there for a very very long time.

To move element next/prev when the cursor hit the boundary of the text was added 5 years ago: https://github.com/davatorium/rofi/commit/c54a817555ed8ac2a2f02805451e7667cbd42d76

**Reply by Asmageddon** [reply]
Wow, I wonder where I had a >5y old config from, considering I first installed rofi 14mo ago, and Arch repos tend to be reasonably up-to-date.

Yeah, in this case my bad - question of whether it's a saner default or not aside - it'd indeed be a breaking change just how today's overdue default config update has been for me.

Thanks for taking the time to chat about this :P

**Reply by DaveDavenport** [reply]
I think with the change in 1.7.X you should be able to unbind `move-char-back` and `move-char-forward` and bind them to `element-next` `element-prev`.. So I think the change you complained about might actually allow you to 'fix' this.

**Reply by Asmageddon** [reply]
Ah, not exactly, I wasn't aware of what the change was, I assumed it was the introduction of moving text cursor that was the change. But yeah, I've been able to configure rofi to my liking, thanks! :)
